src/wui/mapview.cc

```diff
@@ -587,51 +587,46 @@ Widelands::NodeAndTriangle<> MapView::track_sel(const Vector2i& p) {
 }
 
 bool MapView::scroll_map() {
-	// If one of the arrow keys is pressed, scroll this distance
-	uint32_t ScrollDistanceY = g_gr->get_yres() / 4;
-	uint32_t ScrollDistanceX = g_gr->get_xres() / 4;
-
-	if ((get_key_state(SDL_SCANCODE_UP) && get_key_state(SDL_SCANCODE_LEFT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_7) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(-ScrollDistanceX, -ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if ((get_key_state(SDL_SCANCODE_UP) && get_key_state(SDL_SCANCODE_RIGHT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_9) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(ScrollDistanceX, -ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if ((get_key_state(SDL_SCANCODE_DOWN) && get_key_state(SDL_SCANCODE_LEFT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_1) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(-ScrollDistanceX, ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if ((get_key_state(SDL_SCANCODE_DOWN) && get_key_state(SDL_SCANCODE_RIGHT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_3) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(ScrollDistanceX, ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_UP) ||
-	    (get_key_state(SDL_SCANCODE_KP_8) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(0, -ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_DOWN) ||
-	    (get_key_state(SDL_SCANCODE_KP_2) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(0, ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_LEFT) ||
-	    (get_key_state(SDL_SCANCODE_KP_4) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(-ScrollDistanceX, 0), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_RIGHT) ||
-	    (get_key_state(SDL_SCANCODE_KP_6) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(ScrollDistanceX, 0), Transition::Smooth);
-		return true;
-	}
-	return false;
+	// arrow keys
+	const bool kUP = get_key_state(SDL_SCANCODE_UP);
+	const bool kDOWN = get_key_state(SDL_SCANCODE_DOWN);
+	const bool kLEFT = get_key_state(SDL_SCANCODE_LEFT);
+	const bool kRIGHT = get_key_state(SDL_SCANCODE_RIGHT);
+
+	// numpad keys
+	const bool kNumlockOff = !(SDL_GetModState() & KMOD_NUM);
+	const bool kNP1 = get_key_state(SDL_SCANCODE_KP_1);
+	const bool kNP2 = get_key_state(SDL_SCANCODE_KP_2);
+	const bool kNP3 = get_key_state(SDL_SCANCODE_KP_3);
+	const bool kNP4 = get_key_state(SDL_SCANCODE_KP_4);
+	const bool kNP6 = get_key_state(SDL_SCANCODE_KP_6);
+	const bool kNP7 = get_key_state(SDL_SCANCODE_KP_7);
+	const bool kNP8 = get_key_state(SDL_SCANCODE_KP_8);
+	const bool kNP9 = get_key_state(SDL_SCANCODE_KP_9);
+
+	// set the scrolling distance
+	uint8_t denominator =
+	   ((SDL_GetModState() & KMOD_SHIFT) ? 4 : (SDL_GetModState() & KMOD_CTRL) ? 16 : 8);
```

Comment:
Suggestion: Exchange the semantics of Ctrl and Shift (in your gamespeed branch, Shift makes the effect of the keystroke less and Ctrl makes it stronger; here it´s the other way around)

Reply:
Usually we use SHIFT for a big in-/decrease, don't we? So, I would change it in the other branch.

Reply:
We use Shift as a modifier in other parts of the game. Is it used for small or big in-/decreases there?

Reply:
For InputQueueDisplays, Ctrl is the big change (min/max) and Shift is the change of everything by 1. In the attack box Ctrl is also more powerful than Shift. So personally I associate Shift with small and Ctrl with big deltas